### PR TITLE
Merge release notes and changelog.

### LINKED ISSUE #10716: Merge release notes and changelog
### Problem:
There seems to have been a historic seperation between the release notes in the docs folder and the changelog in the root directory of the repo. I double-checked the content and despite it being differently formatted with Restructured Text and Markdown, there isn't enough difference to warrant maintaining two files.

### Expectation:
One file to record development changes that can also be used in user facing documentation.

### Solution:

One option:
1. Move content of `docs/source/release-notes.rst` to `./CHANGELOG.rst`
2. Update `docs/source/release-notes.rst` to use an rST include() to load `CHANGELOG.rst`
3. Delete `./CHANGELOG.md`

There are other options available like:
- setting a symlink from `CHANGELOG.rst` to `docs/source/release-notes.rst` but that is not well-rendered on GitHub.
- switching to [towncrier](https://pypi.org/project/towncrier/)
- switching to [MyST](https://myst-parser.readthedocs.io/) for using Markdown in Sphinx projects (even in parallel to rST files).

### Links

Refs #10715.

### PR DESCRIPTION
**Pull request by jezdez**
Fix #10716.

This introduces MyST (https://myst-parser.readthedocs.io/) to
allow using Markdown files in the documentation and includes
the root level CHANGELOG.md in the releases notes.

### CODE REVIEW

**Comment by jezdez** [comment]
@chenghlee I'm confused why the environment.yml has lost so many packages when I updated it, could you please double-check if I didn't make a mistake somewhere there?

**Comment by jezdez** [comment]
Oh weird, the CI task to test the docs is failing.. investigating.

**Comment by jezdez** [comment]
Okay, I'm not sure why there are both a Conda environment file for documentation AND requirements files. I've updated the latter to install MyST in the CI task.

**Review by chenghlee — APPROVED** [review]
@jezdez: LGTM. The delta in the `utils/environment.yml` package lists looks to be related to changes in conda-forge's packaging structure; none of the removed packages are concerning.

File: docs/source/conf.py [hunk 1]
```diff
@@ -62,8 +78,8 @@
 
 # General information about the project.
 project = 'conda'
-copyright = '2017, Continuum Analytics'
```
Comment (LtDan33) [inline docs/source/conf.py:65]: 👍

File: docs/source/conf.py [hunk 2]
```diff
@@ -62,8 +78,8 @@
...
-author = 'Continuum Analytics'
+copyright = '2017, Anaconda, Inc'
```
Comment (LtDan33) [inline docs/source/conf.py:81]: 👍

**Pull request merged**